### PR TITLE
Add unit tests for mirror service

### PR DESCRIPTION
**Pull request by sethvargo**
Fixes https://github.com/google/exposure-notifications-server/issues/1073

**Release Note**

```release-note
Add unit tests for mirror service
```

### CODE REVIEW

**Comment by sethvargo** [comment]
/assign @mikehelmick 

Please expand all diffs and review carefully. I'm concerned about the logic around ExportRoot and FilenameRoot handling.

File: internal/mirror/mirror.go [hunk 1]
```diff
@@ -238,94 +212,212 @@ func (s *Server) processMirror(ctx context.Context, w http.ResponseWriter, deadl
...
+			MirrorFile: kf,
+		}
+	}
+
+	for i, indexFile := range indexFiles {
+		lastIdx := strings.LastIndex(indexFile, "/")
+		fileName := indexFile[lastIdx+1:]
+
+		// Start ordering a 1 because a value of 0 is also the default value.
+		order := i + 1
```
Comment (mikehelmick) [inline internal/mirror/mirror.go:317]: this is fine, but there shouldn't be any that end up in actions and don't have order set to something.

File: internal/mirror/mirror.go [hunk 2]
```diff
@@ -15,72 +15,31 @@
...
-	for _, zipFile := range zipNames {
-		fullZipFile := fmt.Sprintf("%s%s", mirror.ExportRoot, strings.TrimSpace(zipFile))
-		currentFiles = append(currentFiles, fullZipFile)
-	}
-
-	return currentFiles, nil
-}
+const (
+	mirrorLockPrefix = "mirror-lock"
+	deleteTimeout    = 10 * time.Second
```
[the remark this answers is not in the document]
Comment (sethvargo) [reply internal/mirror/mirror.go]: Done

File: internal/mirror/mirror.go [hunk 3]
```diff
@@ -110,126 +69,141 @@ func (s *Server) handleMirror(ctx context.Context) http.HandlerFunc {
...
+			// If the deadline has passed, do not attempt to process additional
+			// mirrors.
 			if deadlinePassed(deadline) {
 				logger.Warnw("mirror timed out before processing all configurations")
 				break
 			}
 
-			if err := s.processMirror(ctx, w, deadline, mirror); err != nil {
-				logger.Errorw("unable to process mirror", "error", err)
+			if err := s.processMirror(ctx, deadline, mirror); err != nil {
```
[the remark this answers is not in the document]
Comment (sethvargo) [reply internal/mirror/mirror.go:95]: Done. I changed it to return a JSON response that describes each mirror and its status.

**Comment by sethvargo** [comment]
@mikehelmick updated, PTAL

**Review by mikehelmick — APPROVED** [review]

**Pull request merged**